src/prefect/cli/cloud.py

```diff
@@ -35,110 +49,234 @@
 app.add_typer(cloud_app)
 
 
-def build_url_from_workspace(workspace: Dict) -> str:
-    return (
-        f"{PREFECT_CLOUD_API_URL.value()}"
-        f"/accounts/{workspace['account_id']}"
-        f"/workspaces/{workspace['workspace_id']}"
-    )
+# Set up a little API server for browser based `prefect cloud login`
+
+
+def set_login_api_ready_event():
+    login_api.extra["ready-event"].set()
+
+
+login_api = FastAPI(on_startup=[set_login_api_ready_event])
+
+login_api.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+
+class LoginSuccess(BaseModel):
+    api_key: str
+
+
+class LoginFailed(BaseModel):
+    reason: str
+
+
+class LoginResult(BaseModel):
+    type: Literal["success", "failure"]
+    content: Union[LoginSuccess, LoginFailed]
+
+
+class ServerExit(Exception):
+    pass
+
+
+@login_api.post("/success")
+def receive_login(payload: LoginSuccess):
+    login_api.extra["result"] = LoginResult(type="success", content=payload)
+    login_api.extra["result-event"].set()
+
+
+@login_api.post("/failure")
+def receive_failure(payload: LoginFailed):
+    login_api.extra["result"] = LoginResult(type="failure", content=payload)
+    login_api.extra["result-event"].set()
+
+
+async def serve_login_api(cancel_scope):
+    config = uvicorn.Config(login_api, port=3001, log_level="critical")
+    server = uvicorn.Server(config)
+
+    try:
+        await server.serve()
+    except anyio.get_cancelled_exc_class():
+        pass  # Already cancelled, do not cancel again
+    except SystemExit as exc:
+        # If uvicorn is misconfigured, it will throw a system exit and hide the exc
+        app.console.print("[red][bold]X Error starting login service!")
+        cause = exc.__context__  # Hide the system exit
+        traceback.print_exception(type(cause), value=cause, tb=cause.__traceback__)
+        cancel_scope.cancel()
+    else:
+        # Exit if we are done serving the API
+        # Uvicorn overrides signal handlers so without this Ctrl-C is broken
+        cancel_scope.cancel()
 
 
 def confirm_logged_in():
     if not PREFECT_API_KEY:
         profile = prefect.context.get_settings_context().profile
         exit_with_error(
             f"Currently not authenticated in profile {profile.name!r}. "
-            "Please login with `prefect cloud login --key <API_KEY>`."
+            "Please login with `prefect cloud login`."
```

Comment:
```suggestion
            "Please log in with `prefect cloud login`."
```
I know this is pedantic and not consistent with the command but login is the noun and log in is the "verb"